app/service/data_svc.py

```diff
@@ -389,15 +412,23 @@ def __init__(self, dao):
                                                           info['cleanup'].strip().encode(
                                                               'utf-8')).decode() if info.get(
                                                           'cleanup') else None,
-                                                      payload=info.get('payload'), parser=info.get('parser'))
+                                                      payload=info.get('payload'), parser=info.get('parser'),
+                                                      requires_fact_relationship=requires_fact_relationship,
+                                                      creates_fact_relationship=creates_fact_relationship)
                 await self._delete_stale_abilities(ab)
 
-    async def _save_ability_extras(self, identifier, payload, parser):
+    async def _save_ability_extras(self, identifier, payload, parser, requires_fact_relationship, creates_fact_relationship):
```

Comment:
this function seems a little odd. It's like a catch all function for saving random things. I would think on how this can be re organized in a cleaner way